### PR TITLE
[Deprecation] Deprecate sp_weights::OldWeight

### LINKED ISSUE #144: [Deprecation] deprecate "old" weight items
The deprecated items in `frame/support/src/weights.rs` need to be removed

```[tasklist]
### Tasks
- [x] Hard deprecation by adding a warning message
- [x] Remove usage of the deprecated feature in the code base
- [ ] Update examples and tutorials https://github.com/substrate-developer-hub/substrate-docs/issues/2036
- [x] ~Deprecation is noted in the release log~
- [x] ~Removal version is planned~
- [x] Deprecated feature is removed https://github.com/paritytech/substrate/pull/14706
- [ ] Removal of the feature is noted in the release log
```

### PR DESCRIPTION
**Pull request by philoniare**
# Description

*Removes `sp_weights::OldWeight` and its usage*

Fixes #144

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5349134

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5349149

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5349132

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5349286

**Comment by philoniare** [comment]
bot fmt

File: substrate/primitives/weights/src/lib.rs [hunk 1]
```diff
@@ -52,28 +50,6 @@ pub mod constants {
...
-	Copy,
-	RuntimeDebug,
-	Default,
-	MaxEncodedLen,
-	TypeInfo,
-)]
-#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-#[cfg_attr(feature = "serde", serde(transparent))]
-#[deprecated(note = "Will be removed soon; use `Weight` instead.")]
-pub struct OldWeight(pub u64);
```
Comment (ggwpez) [inline substrate/primitives/weights/src/lib.rs:75]: Yea i think that is pretty much it.

It is meant as a beginner issue, so maybe it too easy for you 😆

File: substrate/frame/contracts/src/lib.rs [hunk 2]
```diff
@@ -163,12 +163,6 @@ type DebugBufferVec<T> = BoundedVec<u8, <T as Config>::MaxDebugBufferLen>;
 type EventRecordOf<T> =
 	EventRecord<<T as frame_system::Config>::RuntimeEvent, <T as frame_system::Config>::Hash>;
 
-/// The old weight type.
-///
-/// This is a copy of the [`frame_support::weights::OldWeight`] type since the contracts pallet
-/// needs to support it indefinitely.
-type OldWeight = u64;
```
[the remark this answers is not in the document]
Comment (philoniare) [reply substrate/frame/contracts/src/lib.rs:170]: Got it thanks, will take a look once the CI output is finalized

[the remark this answers is not in the document]
Comment (philoniare) [reply substrate/frame/contracts/src/lib.rs:170]: Just to confirm, you meant that I **should not** touch `contracts` pallet, right?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5354775

File: substrate/frame/contracts/src/lib.rs [hunk 3]
```diff
@@ -627,7 +547,7 @@ pub mod pallet {
 		/// To avoid this situation a constructor could employ access control so that it can
 		/// only be instantiated by permissioned entities. The same is true when uploading
 		/// through [`Self::instantiate_with_code`].
-		#[pallet::call_index(3)]
+		#[pallet::call_index(0)]
```
Comment (xlc) [inline substrate/frame/contracts/src/lib.rs:550]: you should never change index

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5355068

**Comment by philoniare** [comment]
bot fmt

File: substrate/frame/contracts/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ggwpez) [reply substrate/frame/contracts/src/lib.rs]: yep

File: substrate/frame/contracts/src/lib.rs (hunk 3 above)
Comment (ggwpez) [reply substrate/frame/contracts/src/lib.rs]: yea please revert any changes to the contracts logic and just copy&paste the `OldWeight` type into it.

Comment (philoniare) [reply substrate/frame/contracts/src/lib.rs]: yep, that's already done. Can you run the `bot fmt`? I'm unable to run it because I'm not a member.

Comment (philoniare) [reply substrate/frame/contracts/src/lib.rs]: actually, let me just close this one and create a new PR, this might be easier.

Comment (xlc) [reply substrate/frame/contracts/src/lib.rs]: call index determines the encoded value of the extrinsic. change index means a tx generated with old runtime will be decoded to a different extrinsic and that’s a security risk. that’s why we always explicitly annotate the call index and should never change them to keep compatibility. 
you also deleted some extrinsic and that means they won’t be available anymore. and again that’s a breaking change which requires bit more process (eg announce it to ensure ecosystem tools are all updated)
so to keep the scope of this particular PR minimal and safe you from doing extra process work, best to keep the pallet contract intact

Comment (philoniare) [reply substrate/frame/contracts/src/lib.rs]: thank you, that makes sense. Appreciate you taking the time to share your valuable knowledge

**Pull request closed without merging**